### PR TITLE
Angry bee behaviour

### PR DESCRIPTION
**Pull request by MikkMunk**
Still needs animation and hurt effect interaction

### CODE REVIEW

**Review by latedude2 — CHANGES_REQUESTED** [review]
YOU FORGOT TO REMOVE MAGIC NUMBERS - Olga

File: Assets/Scripts/BeeAngryController.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public int minDist = 2;
+
+    private NavMeshAgent agent;
+    public Animator anim;
+    private int refreshTargetTimer = 0;
+    public int refreshTargetTimerLimit = 50;
+
+    void Start()
+    {
+        anim = GetComponentInChildren<Animator>();
```
Comment (Jokubas126) [inline Assets/Scripts/BeeAngryController.cs]: Just better coding practice and better for readability
```suggestion
        animator = GetComponentInChildren<Animator>();
```

File: Assets/Scripts/BeeAngryController.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            agent.destination = player.position;
+            refreshTargetTimer = refreshTargetTimerLimit;
+        }
+
+        //add timer
+        if (dist <= minDist)
+        {
+            BeeAttack();
+        }
+    }
```
Comment (Jokubas126) [inline Assets/Scripts/BeeAngryController.cs]: can we put those separate parts of Update function to smaller functions? Would be easier to understand which part does what

File: Assets/Scripts/EndTaskInteraction.cs [hunk 3]
```diff
@@ -95,6 +98,18 @@ void FixedUpdate()
...
+            if (!angryBeeSpawned)
+            {
+                //Spawn bee at position and set player for targetting to be the player object
+                GameObject bee = Instantiate(angryBee, beeSpawnPoint.transform.position, Quaternion.identity);
+                bee.GetComponent<BeeAngryController>().player = gameObject.transform.parent.gameObject.transform;
+
+                //apply the bee object to the endtest script in the parent object, so we can remove when we reach the cave
+                gameObject.transform.parent.gameObject.GetComponent<EndTest>().angryBee = bee;
+
+                angryBeeSpawned = true;
```
Comment (Jokubas126) [inline Assets/Scripts/EndTaskInteraction.cs]: Just better code practice
```suggestion
                hasAngryBeeSpawned = true;
```

File: Assets/Scripts/BeeAngryController.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        agent = GetComponent<NavMeshAgent>();
+
+        anim.Play("AngryFlight");
+
+        agent.stoppingDistance = 2;
+    }
+
+    void Update()
+    {
+        float dist = Vector3.Distance(player.position, transform.position);
```
Comment (Jokubas126) [inline Assets/Scripts/BeeAngryController.cs]: Please rename it to something that would tell what "distance" is it. Like `agentDistance`, `relativeDistance`, `distanceToPlayer` or something like that

File: Assets/Scripts/BeeAngryController.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        agent.speed = 25 * (dist * 0.01f);
+        if (agent.speed < 2)
+            agent.speed = 2;
+
+        //we set destination for target to run less than every frame, cause it's heavy over longer distances
+        if (refreshTargetTimer <= 0)
+        {
+            agent.destination = player.position;
+            refreshTargetTimer = refreshTargetTimerLimit;
+        }
```
Comment (Jokubas126) [inline Assets/Scripts/BeeAngryController.cs:40]: Can we do it on a separate thread if it's heavy as well? `StartCoroutine(IEnumerator coroutine);` should do the job

File: Assets/Scripts/BeeAngryController.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    void BeeAttack()
+    {
+        //Set bee run attack animation here
+        anim.Play("Attack");
+
+        //player.GetComponent<HurtEffect>().Hit();
+        Debug.Log("DEAD!");
```
Comment (Jokubas126) [inline Assets/Scripts/BeeAngryController.cs]: It doesn't work yet or did you forget to uncoment it?

Comment (Jokubas126) [reply Assets/Scripts/BeeAngryController.cs]: oh I saw the PR description. It's ok 😄

File: Assets/Scripts/BeeAngryController.cs (hunk 5 above)
Comment (Jokubas126) [reply Assets/Scripts/BeeAngryController.cs:40]: Some examples here: [https://docs.unity3d.com/ScriptReference/MonoBehaviour.StartCoroutine.html](https://docs.unity3d.com/ScriptReference/MonoBehaviour.StartCoroutine.html)

**Review by Jokubas126 — APPROVED** [review]
Beautiful ❤️

**Review by latedude2 — APPROVED** [review]

**Pull request merged**